pytorch_tabnet/abstract_model.py

```diff
@@ -320,177 +315,366 @@ def load_model(self, filepath):
 
         self.__init__(**loaded_params)
 
-        self.init_network(
-            input_dim=self.input_dim,
-            output_dim=self.output_dim,
-            n_d=self.n_d,
-            n_a=self.n_a,
-            n_steps=self.n_steps,
-            gamma=self.gamma,
-            cat_idxs=self.cat_idxs,
-            cat_dims=self.cat_dims,
-            cat_emb_dim=self.cat_emb_dim,
-            n_independent=self.n_independent,
-            n_shared=self.n_shared,
-            epsilon=self.epsilon,
-            virtual_batch_size=1024,
-            momentum=self.momentum,
-            device_name=self.device_name,
-            mask_type=self.mask_type
-        )
+        self._set_network()
         self.network.load_state_dict(saved_state_dict)
         self.network.eval()
         return
 
-    def fit_epoch(self, train_dataloader, valid_dataloader):
+    def _train_epoch(self, train_loader):
         """
-        Evaluates and updates network for one epoch.
+        Trains one epoch of the network in self.network
 
         Parameters
         ----------
-            train_dataloader: a :class: `torch.utils.data.Dataloader`
+            train_loader: a :class: `torch.utils.data.Dataloader`
                 DataLoader with train set
-            valid_dataloader: a :class: `torch.utils.data.Dataloader`
-                DataLoader with valid set
         """
-        train_metrics = self.train_epoch(train_dataloader)
-        valid_metrics = self.predict_epoch(valid_dataloader)
+        self.network.train()
 
-        fit_metrics = {'train': train_metrics,
-                       'valid': valid_metrics}
+        list_y_true = []
+        list_y_score = []
 
-        return fit_metrics
+        for batch_idx, (X, y) in enumerate(train_loader):
+            self._callback_container.on_batch_begin(batch_idx)
 
-    @abstractmethod
-    def train_epoch(self, train_loader):
-        """
-        Trains one epoch of the network in self.network
+            batch_outs, batch_logs = self._train_batch(X, y)
 
-        Parameters
-        ----------
-            train_loader: a :class: `torch.utils.data.Dataloader`
-                DataLoader with train set
-        """
-        raise NotImplementedError('users must define train_epoch to use this base class')
+            self._callback_container.on_batch_end(batch_idx, batch_logs)
 
-    @abstractmethod
-    def train_batch(self, data, targets):
+            list_y_true.append(y)
+            list_y_score.append(batch_outs["scores"])
+
+        y_true, scores = self.stack_batches(list_y_true, list_y_score)
```

Comment:
During train_epoch, I suggest than we don't store nor stack the batch predictions but only compute the average loss over batches.
If one wants to monitor the early stopping metric on training_set one should add `(X_train, y_train)` to the eval set, this would save some time and some memory when the training metric is not important and should also avoid memory error when switching to out of memory files.

Reply:
Yes, agree, done